Modules/Server/Sources/PocketCastsServer/Public/Sync/SyncTask+ServerChanges.swift

```diff
@@ -315,4 +329,77 @@ extension SyncTask {
     func isPlayerPlaying(episode: Episode) -> Bool {
         ServerConfig.shared.playbackDelegate?.isActivelyPlaying(episodeUuid: episode.uuid) ?? false
     }
+
+    func importBookmark(_ apiBookmark: Api_SyncUserBookmark) async {
+        let bookmarkManager = dataManager.bookmarks
```

Comment:
Not for this PR, but for the future: I wonder if we can decouple the model/database layer from the syncing logic.

I believe we can make changes in a way that the `SyncTask` doesn't need to know anything about the returned model and our stored model. What we can try to do is to move all the "BookmarkManager` logic to the `Bookmark` itself.

A few ideas we could potentially explore:

* A Bookmark can be created from a `Api_SyncUserBookmark`. It would also be able to check if it already exists or not;
* The bookmark itself has `.save()` and `.delete()` methods
* `save()` would be smart enough to know if the Bookmark needs to be added or updated on the DB layer

This way, all this function would do could be something like:

```swift
let bookmark = Bookmark(from: apiBookmark)
apiBookmark.shouldDelete ? bookmark.permanentlyDelete() : bookmark.save()
```

Reply:
I think that would be a nice change from the way the app currently works. I think if we do this we'll need to make some bigger changes to allow our future code to also use the same patterns. Jumping off from your idea, I'm thinking:

- Create a parent `DataModel` class that contains all the base functionality for save, deleted, create, etc. 
    - It would also define the base initializer to take a DataManager and set its internal property
- Bookmarks, and other models would subclass off of the parent and add override the methods to add their own functionality.